src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5429,9 +5438,27 @@ public override BoundNode VisitDiscardExpression(BoundDiscardExpression node)
 
         public override BoundNode VisitThrowExpression(BoundThrowExpression node)
         {
-            var result = base.VisitThrowExpression(node);
-            SetUnknownResultNullability();
-            return result;
+            VisitThrow(node.Expression);
+            return null;
+        }
+
+        public override BoundNode VisitThrowStatement(BoundThrowStatement node)
+        {
+            VisitThrow(node.ExpressionOpt);
+            return null;
+        }
+
+        private void VisitThrow(BoundExpression expr)
+        {
+            var result = VisitRvalueWithResult(expr);
+            if ((expr?.ConstantValue?.IsNull == true && !expr.IsSuppressed) ||
```

Comment:
>!expr.IsSuppressed [](start = 56, length = 18)

Why ```!expr.IsSuppressed``` check is combined only with the constant check, is shouldn't is suppress the warning for the conditions below as well?

Reply:
With a typed expression, the `result` is marked as non-nullable when there is a suppression. 
But with a `null` typed exception, we don't do that (the result remains `default`). I don't know of any other case like this specific one  (`throw null`), so I think it's better to specially handle here.